sendtx/tx_sender.go

```diff
@@ -381,12 +468,59 @@ func (txSnd *TxSender) populateTxBuilder(
 		return nil, err
 	}
 
+	builder.SetProtocolParameters(protocolParams)
+
+	srcMinUtxo := srcConfig.MinUtxoValue
+
+	srcNativeTokens := getNativeTokensFromUtxos(utxos)
+
+	// calculate minUtxo for change output
+	if len(srcNativeTokens) > 0 {
+		potentialTokenCost, err := cardanowallet.GetTokenCostSum(builder,
+			senderAddr, []cardanowallet.Utxo{
+				{
+					Amount: outputCurrencyLovelace,
+					Tokens: srcNativeTokens,
+				},
+			},
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		srcMinUtxo = max(srcMinUtxo, potentialTokenCost)
+	}
+
+	// calculate minUtxo for multisig output
+	if outputNativeToken > 0 {
+		nativeTokenSrc, err := cardanowallet.NewTokenWithFullName(srcNativeTokenFullName, true)
+		if err != nil {
+			return nil, err
+		}
+
+		potentialTokenCost, err := cardanowallet.GetTokenCostSum(builder,
+			receiverAddr, []cardanowallet.Utxo{
+				{
+					Amount: outputCurrencyLovelace,
+					Tokens: []cardanowallet.TokenAmount{
+						cardanowallet.NewTokenAmount(nativeTokenSrc, outputNativeToken),
+					},
+				},
+			},
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		outputCurrencyLovelace = max(outputCurrencyLovelace, potentialTokenCost)
```

Comment:
we need to return this value also because of
```
inputsSumCurrencyLovelace := inputsSum[cardanowallet.AdaTokenName]
	change := inputsSumCurrencyLovelace - outputCurrencyLovelace - feeCurrencyLovelace
	// handle overflow or insufficient amount
	if change != 0 && (change > inputsSumCurrencyLovelace || change < srcConfig.MinUtxoValue) {
		return []byte{}, "", fmt.Errorf("insufficient amount %d for %d or min utxo not satisfied",
			inputsSumCurrencyLovelace, outputCurrencyLovelace+feeCurrencyLovelace)
	}

	if change != 0 {
		builder.UpdateOutputAmount(-1, change)
	} else {
		builder.RemoveOutput(-1)
	}
```